src/data/finset/pointwise.lean

```diff
@@ -985,4 +985,66 @@ lemma smul_finset_univ₀ [fintype β] (ha : a ≠ 0) : a • (univ : finset β)
 coe_injective $ by {  push_cast, exact set.smul_set_univ₀ ha }
 
 end group_with_zero
+
+section smul_with_zero
+variables [has_zero α] [has_zero β] [smul_with_zero α β] [decidable_eq β] {s : finset α}
+  {t : finset β}
+
+/-!
+Note that we have neither `smul_with_zero α (finset β)` nor `smul_with_zero (finset α) (finset β)`
+because `0 * ∅ ≠ 0`.
+-/
+
+lemma smul_zero_subset (s : finset α) : s • (0 : finset β) ⊆ 0 := by simp [subset_iff, mem_smul]
+lemma zero_smul_subset (t : finset β) : (0 : finset α) • t ⊆ 0 := by simp [subset_iff, mem_smul]
+
+lemma nonempty.smul_zero (hs : s.nonempty) : s • (0 : finset β) = 0 :=
+s.smul_zero_subset.antisymm $ by simpa [mem_smul] using hs
```

Comment:
We should have a tactic that adds all these versions using `coe_injective` and `set` lemmas. But this definitely out of the scope for this PR.

Reply:
Agreed! Now that I somehow understand metaprogramming I might try to write it.